src/servers/grpc_server.cc

```diff
@@ -57,18 +57,19 @@ namespace {
 class AsyncResources : public nvrpc::Resources {
  public:
   explicit AsyncResources(
-      InferenceServer* server, int infer_threads, int mgmt_threads)
+      const std::shared_ptr<InferenceServer>& server, int infer_threads,
+      int mgmt_threads)
       : m_Server(server), m_MgmtThreadPool(mgmt_threads),
         m_InferThreadPool(infer_threads)
   {
   }
 
-  InferenceServer* GetServer() { return m_Server; }
+  const std::shared_ptr<InferenceServer>& GetServer() { return m_Server; }
   ThreadPool& GetMgmtThreadPool() { return m_MgmtThreadPool; }
   ThreadPool& GetInferThreadPool() { return m_InferThreadPool; }
 
  private:
-  InferenceServer* m_Server;
+  std::shared_ptr<InferenceServer> m_Server;
```

Comment:
question about consistency in style here. Shouldn't this be `server_` and not `m_Server`?

Reply:
Yes, those names are not in the correct style.